### PR TITLE
issue/31 - FES Upload Field

### LINKED ISSUE #31: File uploads in FES
At this time, the S3 extension works with FES but only applies to the Downloads Prices and Files field in the submission form. I think enough customers are requesting that S3 support extend, at least optionally, to other file upload fields. Here are the options I see:

A. Add an option in the File Upload field settings for putting the files on S3 instead of local
B. Create a new S3 Upload field which will exist in the Add Extension Created Fields section of the submission form builder
C. Create a new extension which does one of the following if this isn't deemed appropriate for inclusion in the S3 extension
D. If we really can't support anything else, at least write up a little more of an explanation in our docs or in a canned reply explaining why as it isn't clear to customers why only one field is supported

For reference, here's a screenshot of the submission form:
screenshot 2016-06-21 14 27 42

### PR DESCRIPTION
**Pull request by sunnyratilal**
Resolves #31

### CODE REVIEW

File: edd_s3.php [hunk 1]
```diff
@@ -110,48 +119,49 @@ private function __construct() {
...
 
 		$this->constants();
 		$this->load_textdomain();
 		$this->init();
 
-		$this->s3 = new \Aws\S3\S3MultiRegionClient( array(
-			'version'     => '2006-03-01',
-			'credentials' => array(
+		$this->s3 = new S3MultiRegionClient( array(
+			'version'           => '2006-03-01',
```
Comment (pippinsplugins) [inline edd_s3.php]: Is this version right?

File: includes/class-fes-field.php [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+			$readonly = $this->readonly;
+		}
+
+		/**
+		 * Filter the User ID.
+		 *
+		 * @param int $user_id User ID.
+		 * @param int $id      Field ID.
+		 */
+		$user_id = apply_filters( 'fes_render_edd_s3_field_user_id_admin', $user_id, $this->id );
```
Comment (pippinsplugins) [inline includes/class-fes-field.php]: This filter name mentions `admin` but is on the frontend.

File: includes/class-fes-field.php [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+
+		/**
+		 * Filter the `readonly` value.
+		 *
+		 * @param bool $readonly Whether the field is readonly or not.
+		 * @param int  $user_id  User ID.
+		 * @param int  $id       Field ID.
+		 */
+		$readonly = apply_filters( 'fes_render_edd_s3_field_readonly_admin', $readonly, $user_id, $this->id );
```
Comment (pippinsplugins) [inline includes/class-fes-field.php]: This one also mentions `admin`.

**Comment by pippinsplugins** [comment]
@sunnyratilal we have a merge conflict that needs to be resolved.

File: edd_s3.php (hunk 1 above)
Comment (sunnyratilal) [reply edd_s3.php]: Yep that's the latest version, I can change the string to `latest` though.

**Pull request merged**